src/store/courses/thunk.js

```diff
@@ -0,0 +1,176 @@
+				COURSES_ALL_RESPONSE_MESSAGES.default
+		);
+		dispatch(setCoursesAction([]));
+	} finally {
+		changeIsLoading && changeIsLoading(false);
+	}
+};
+
+export const fetchAddCourse = (course, changeIsLoading) => async (dispatch) => {
+	const result = { successful: false };
+
+	try {
+		changeIsLoading(true);
+
+		const response = await addCourse(course);
+		const { data } = response;
+
+		if (response.status === 201 && data.successful) {
+			const courseObj = data.result;
+
+			dispatch(addCourseAction(courseObj));
+			result.successful = true;
+			toast.success(COURSES_ADD_RESPONSE_MESSAGES[201]);
+		} else {
+			toast.error(
+				COURSES_ADD_RESPONSE_MESSAGES[response.status] ??
+					COURSES_ADD_RESPONSE_MESSAGES.default
+			);
+		}
+	} catch (error) {
+		toast.error(
+			COURSES_ADD_RESPONSE_MESSAGES[error.response?.status] ??
+				COURSES_ADD_RESPONSE_MESSAGES.default
+		);
+	} finally {
+		changeIsLoading(false);
+	}
+
+	return result;
+};
+
+export const fetchUpdateCourse =
+	(courseId, course, changeIsLoading) => async (dispatch) => {
+		const result = { successful: false };
+
+		try {
+			changeIsLoading(true);
+
+			const response = await updateCourse(courseId, course);
+			const { data } = response;
+
+			if (response.status === 200 && data.successful) {
+				const courseObj = data.result;
+
+				dispatch(updateCourseAction(courseObj));
+				result.successful = true;
+				toast.success(COURSES_UPDATE_RESPONSE_MESSAGES[200]);
+			} else {
+				toast.error(
+					COURSES_UPDATE_RESPONSE_MESSAGES[response.status] ??
+						COURSES_UPDATE_RESPONSE_MESSAGES.default
+				);
+			}
+		} catch (error) {
+			toast.error(
+				COURSES_UPDATE_RESPONSE_MESSAGES[error.response?.status] ??
+					COURSES_UPDATE_RESPONSE_MESSAGES.default
+			);
+		} finally {
+			changeIsLoading(false);
+		}
+
+		return result;
+	};
+
+export const fetchDeleteCourse =
+	(courseId, changeIsLoading) => async (dispatch) => {
+		try {
+			changeIsLoading(true);
+
+			const response = await deleteCourse(courseId);
+			const { data } = response;
+
+			if (response.status === 200 && data.successful) {
+				dispatch(deleteCourseAction(courseId));
+				toast.success(COURSES_DELETE_RESPONSE_MESSAGES[200]);
+			} else {
+				toast.error(
+					COURSES_DELETE_RESPONSE_MESSAGES[response.status] ??
+						COURSES_DELETE_RESPONSE_MESSAGES.default
+				);
+			}
+		} catch (error) {
+			toast.error(
+				COURSES_DELETE_RESPONSE_MESSAGES[error.response?.status] ??
+					COURSES_DELETE_RESPONSE_MESSAGES.default
+			);
+		} finally {
+			changeIsLoading(false);
+		}
+	};
+
+export const fetchCourse = (courseId, changeIsLoading) => async () => {
+	const result = { successful: false, data: null };
+
+	try {
+		changeIsLoading(true);
+
+		const response = await getCourse(courseId);
+		const { data } = response;
+
+		if (response.status === 200 && data.successful) {
+			const courseObj = data.result;
+			result.successful = true;
+			result.data = courseObj;
```

Comment:
Just FYI
Here is a useful trick if you want conditionally create object with come new field and not mutate it.
Can be useful in future.
const courseObj = {
...data.result,
...(response.status === 200 && data.successful) && {successful: true, ... etc})
}

Reply:
Done